### DISCUSSION TITLE
API Endpoint: IsHodling

### DISCUSSION THREAD

**Discussion by mubashariqbal** [deso-protocol/dips#4] [Medium CIPs]
## One-Line Description

Add a new endpoint that returns the creator coin holding information of one user by another user

## Explanation and Motivation

Many application need to know if a user is holding the creator coin of another user, to be able to gate access to content or features.

To enable this currently you need to download the complete list of holders for a user and then check if the specific person is in that list.

This can be time consuming and a heavy process for popular users. It can also cause a delay if the transaction is recently conducted.

**Comment by lazynina** [comment]
I think applications using this endpoint may also want to know how much of a given creator a user is hodling. For example, users must hold at least .1 LazyNina to access some content.

**Reply by chafreaky** [reply]
As well as when (or block height) the latest transaction for that specific creator coin was made.
Useful for opening up staking to access content (e.g. you must have been a hodler for at least 1 month to view this).

**Reply by mubashariqbal** [reply]
@lazynina: Agreed.  I was planning on returning a BalanceEntryResponse object if the user is holding the specified creators coin, in addition to the true/false flag.

@chafreaky I agree this would be useful, but I don't think that would easy to find in the current core setup. I suggest we keep this CIP simple for this initial change.  Then later when the backend moves to Postgres or some other easy way to search the transactions we can add additional fields of data.

**Reply by lazynina** [reply]
I agree with @mubashariqbal. The current setup isn't conducive to what you've suggested @chafreaky.

I believe at @maebeam is working on some changes to our transaction index which would make this hodling time computation easier in the future.

However, you probably need to know more than just the length of time I've held, but rather my balance after each creator coin transaction (buy/sell/transfer), in order to do useful gating of content.

**Comment by mubashariqbal** [comment]
@lazynina Do I have approval for this change? I think that's need for me to officially more forward.

**Reply by lazynina** [reply]
You've got my approval.  Please open an issue on the backend repo, assign it to yourself, and add it to the [medium](https://github.com/orgs/bitclout/projects/2) board please

**Reply by mubashariqbal** [reply]
Thanks, will do!

**Reply by mubashariqbal** [reply]
@lazynina FYI: I added the issue (https://github.com/bitclout/backend/issues/75) but doesn't look like I have necessary permissions to assign it to myself or add it to the board.

**Reply by maebeam** [reply]
Congrats! I created a new "approved" label and added it to this discussion.

**Comment by tijno** [comment]
Is this the first approved ? Nice work - cant wait for the "you hold this coin" icon on creators profile card.

**Reply by mubashariqbal** [reply]
> Is this the first approved ? Nice work - cant wait for the "you hold this coin" icon on creators profile card.

I believe so!  So many uses for this data, excited to get it built.

**Comment by mubashariqbal** [comment]
PR submitted: https://github.com/bitclout/backend/pull/77
